src/mixins/object_geometry.mixin.js

```diff
@@ -643,18 +643,22 @@
 
     /**
      * calculate rotation matrix of an object
+     * @param {boolean} [absolute] true means angle is measured relative to canvas, false means angle is measured relative to parent
      * @return {Array} rotation matrix for the object
      */
-    _calcRotateMatrix: function() {
-      return util.calcRotateMatrix(this);
+    _calcRotateMatrix: function (absolute, accountForFlipping) {
+      var angle = absolute ? this.getTotalAngle() : this.angle;
+      accountForFlipping && this.flipX && (angle -= 180);
```

Comment:
i wonder how nested flippings behave.

Reply:
I wonder/forgot why I accountd for flipX only for nested objects, why not for all objects?

Reply:
It's bad. Good catch.
Isn't flipX/flipY a z plane transformation? If so wouldn't it be better to multiply 3d matrices?

Reply:
rotation as used by fabric is a 2d plane transformation. So it shouldn't relate to z plane transformations. I think this is the cause of the problem

Reply:
What about [`ml-matrix`](https://mljs.github.io/matrix/)?
https://mljs.github.io/matrix/classes/QrDecomposition.html

Reply:
https://developer.mozilla.org/en-US/docs/Web/API/WebGL_API/Matrix_math_for_the_web

Reply:
![image](https://user-images.githubusercontent.com/34343793/166204868-bc145cda-5435-4850-ade0-46ed148fab7f.png)

![image](https://user-images.githubusercontent.com/34343793/166205014-a959b0a3-dce7-42eb-b82f-23e0c7ade658.png)

Reply:
@asturur Thoguhts?

Reply:
https://codepen.io/ktamas77/pen/AXmmeX?editors=0010

Reply:
I guess we can use matrix projection from 3d -> 2d...

Reply:
But it open up 3d and I'm not sure we want that.
So I vote to remove flip props as they are too broken

Reply:
https://en.wikipedia.org/wiki/Orthographic_projection